lib/Backends/Backend.cpp

```diff
@@ -40,3 +42,67 @@ void Backend::optimizeFunction(CompilationMode mode, Function *F) {
     ::glow::optimize(F, mode);
   }
 }
+
+TraceInfo Backend::autoInstrument(IRFunction *IR,
+                                  size_t traceEventDataSize) const {
+  if (traceEventDataSize == 0) {
+    return TraceInfo(false, traceEventDataSize);
+  }
+
+  Function *F = IR->getGraph();
+  // Get all instructions in the IRFunction.
+  IRFunction::InstListTy &instructions = IR->getInstrs();
+
+  // Build a placeholder to a backing tensor with space to fit all timestamps.
+  auto *backingPH = F->getParent()->createPlaceholder(
+      ElemKind::Int64ITy, {instructions.size() + 1, traceEventDataSize},
```

Comment:
I'm confused by the dimensions here -- why is the second dimension here `traceEventDataSize`?

Reply:
Oh. The intention was for to be essentially an array of instructions+1 elements of traceEventDataSize bytes but then I changed the type to int64 and didn't update the size, so this is 8x what we need.